public/app/core/services/backend_srv.ts

```diff
@@ -142,6 +142,56 @@ export class BackendSrv implements BackendService {
     });
   }
 
+  chunked(options: BackendSrvRequest): Observable<FetchResponse<Uint8Array | undefined>> {
+    const controller = new AbortController();
+    const requestId = options.requestId ?? `chunked-${Date.now()}`;
+    return new Observable((observer) => {
+      let disconnect = false;
+      const sub = this.fetch<ReadableStream<Uint8Array>>({
+        ...options,
+        requestId,
+        responseType: 'body',
+        showErrorAlert: false,
+        showSuccessAlert: false,
+        hideFromInspector: true,
+        abortSignal: controller.signal,
+      }).subscribe((result) => {
+        const reader = result.data.getReader();
+        async function process() {
+          while (!disconnect) {
+            try {
+              const chunk = await reader.read();
+              observer.next({
+                ...result,
+                data: chunk.value,
+              });
+              if (chunk.done) {
+                break;
+              }
```

Comment:
```suggestion
           if (chunk.done) {
                break;
              }
              observer.next({
                ...result,
                data: chunk.value,
              });
```
Should these be switched in case there's no value already from the beginning?

Reply:
The value may be empty -- that is why the type is: `<Uint8Array | undefined>`